code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -0,0 +1,420 @@
+	cost = 1
+	/// Whether we've done the ritual. Only doable once.
+	var/was_completed = FALSE
+
+/datum/heretic_knowledge/knowledge_ritual/New()
+	. = ..()
+	var/static/list/potential_organs = list(
+		/obj/item/organ/appendix,
+		/obj/item/organ/tail,
+		/obj/item/organ/eyes,
+		/obj/item/organ/tongue,
+		/obj/item/organ/ears,
+		/obj/item/organ/heart,
+		/obj/item/organ/liver,
+		/obj/item/organ/stomach,
+		/obj/item/organ/lungs,
+	)
+
+	var/static/list/potential_easy_items = list(
+		/obj/item/shard,
+		/obj/item/candle,
+		/obj/item/book,
+		/obj/item/pen,
+		/obj/item/paper,
+		/obj/item/toy/crayon,
+		/obj/item/flashlight,
+		/obj/item/clipboard,
+	)
+
+	var/static/list/potential_uncommoner_items = list(
+		/obj/item/restraints/legcuffs/beartrap,
+		/obj/item/restraints/handcuffs/cable/zipties,
+		/obj/item/circular_saw,
+		/obj/item/scalpel,
+		/obj/item/binoculars,
+		/obj/item/clothing/gloves/color/yellow,
+		/obj/item/melee/baton/security,
+		/obj/item/clothing/glasses/sunglasses,
+	)
+
+	required_atoms = list()
+	// 2 organs. Can be the same.
+	required_atoms[pick(potential_organs)] += 1
+	required_atoms[pick(potential_organs)] += 1
+	// 2-3 random easy items.
+	required_atoms[pick(potential_easy_items)] += rand(2, 3)
+	// 1 uncommon item.
+	required_atoms[pick(potential_uncommoner_items)] += 1
+
+/datum/heretic_knowledge/knowledge_ritual/on_research(mob/user)
+	. = ..()
+
+	var/list/requirements_string = list()
+
+	to_chat(user, span_hierophant("The [name] requires the following:"))
+	for(var/obj/item/path as anything in required_atoms)
+		var/amount_needed = required_atoms[path]
+		to_chat(user, span_hypnophrase("[amount_needed] [initial(path.name)][amount_needed == 1 ? "":"s"]..."))
+		requirements_string += "[amount_needed == 1 ? "":"[amount_needed] "][initial(path.name)][amount_needed == 1 ? "":"s"]"
+
+	to_chat(user, span_hierophant("Completing it will reward you [KNOWLEDGE_RITUAL_POINTS] knowledge points. You can check the knowledge in your Researched Knowledge to be reminded."))
+
+	desc = "Allows you to transmute [english_list(requirements_string)] for [KNOWLEDGE_RITUAL_POINTS] bonus knowledge points. This can only be completed once."
+
+/datum/heretic_knowledge/knowledge_ritual/recipe_snowflake_check(mob/living/user, list/atoms, list/selected_atoms, turf/loc)
+	return !was_completed
+
+/datum/heretic_knowledge/knowledge_ritual/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
+	var/datum/antagonist/heretic/our_heretic = IS_HERETIC(user)
+	our_heretic.knowledge_points += KNOWLEDGE_RITUAL_POINTS
+	was_completed = TRUE
+
+	var/drain_message = pick(strings(HERETIC_INFLUENCE_FILE, "drain_message"))
+	to_chat(user, span_boldnotice("[name] completed!"))
+	to_chat(user, span_hypnophrase(span_big("[drain_message]")))
+	desc += " (Completed!)"
+	return TRUE
+
+#undef KNOWLEDGE_RITUAL_POINTS
+
+/*
+ * The special final tier of knowledges that unlocks ASCENSION.
+ */
+/datum/heretic_knowledge/final
```

Comment:
i believe you mentioned somewhere that you wanted some ideas about the last objective before ascensions, my suggestions:
 - Witnessing : to complete your ascension ritual you will have to be seen by at least X *living concious people*.
 - Glory of the wound: to complete your ascension ritual you have to have at least X wounds of Y degree (they get healed or sth)
 - Emperor's presence: have at least X minions summoned (flesh exclusive?)
 - Demand for corruption: have at least X rusted tiles (rust exclusive?)
 - One with the pyre: burn for at least X seconds prior to ascensions(or require to stand in a pyre for X seconds before ascending) (ash exclusive?)
 - Meeting with nobility: prepare a feast for the aristocrat? (require a table, 2 chairs, and a dining set?) (void exclusive)

Reply:
These are great ideas, i'll look about putting some in